platforms/android/tangram/src/main/java/com/mapzen/tangram/MapRenderer.java

```diff
@@ -34,29 +41,41 @@ public void onDrawFrame(final GL10 gl) {
 
         boolean mapViewComplete;
         boolean isCameraEasing;
+        boolean isAnimating;
 
         synchronized(map) {
-            mapViewComplete = nativeUpdate(mapPointer, delta);
-            isCameraEasing = nativeRender(mapPointer);
+            int state = nativeUpdate(mapPointer, delta);
+
+            nativeRender(mapPointer);
+
+            mapViewComplete = (state == VIEW_COMPLETE);
+            isCameraEasing = (state & VIEW_CHANGING) != 0;
+            isAnimating = (state & VIEW_ANIMATING) != 0;
         }
 
         if (isCameraEasing) {
-            uiThreadHandler.post(setMapRegionAnimatingRunnable);
+            if (!isPrevCameraEasing) {
+                uiThreadHandler.post(setMapRegionAnimatingRunnable);
+            }
         } else if (isPrevCameraEasing) {
             uiThreadHandler.post(setMapRegionIdleRunnable);
         }
 
-        boolean viewComplete = mapViewComplete && !isPrevMapViewComplete;
+        if (isAnimating) {
+            map.requestRender();
```

Comment:
It seems simpler to apply the re-rendering logic in the core, rather than duplicating it for each of the platforms. Why this change?

Reply:
It's mostly beneficial on Android to safe one native-to-java call. Though to me this seems also is more clean - or otherwise name it `updateAndRequestRerenderAsNeeded()`

Reply:
Fair enough!